packages/ra-core/src/core/useFirstResourceWithListAccess.tsx

```diff
@@ -0,0 +1,20 @@
+import { ReactElement } from 'react';
+import { useCanAccessResources } from '../auth/useCanAccessResources';
+
+/**
+ * A hook that returns the first resource users have list access to.
+ * It calls the `authProvider.canAccess` if available to check the permissions.
+ */
+export const useFirstResourceWithListAccess = (resources: ReactElement[]) => {
```

Comment:
to be revisited as we now check access control in the oage controllers and not in the Resources anymore

Reply:
This has nothing to do with the Resource component. It's used in `<CoreAdminRoutes>`